lib/server-starter.js

```diff
@@ -90,7 +92,23 @@ class Server extends EventEmitter {
     });
 
     // Should be resolved by the "spawn" event, but that is only supported in Node 15+
-    return Promise.resolve;
+    if (process.platform !== 'win32') return Promise.resolve;
+    return new Promise((resolve) => {
+      const retryTime = 60;
+      const now = new Date();
+      const timeToStop = new Date(now.getTime() + win32Timeout);
+      const port = this.port;
+      (function loop () {
+        const connection = net.connect(port, resolve);
+        connection.on('error', err => {
+          if (err.code === 'ECONNREFUSED') {
```

Comment:
And if it's something else?

Reply:
should resolve the promise. But it actually doesn't, my mistake, see my answer about the logic a few minutes ago

Reply:
It should reject the promise, not resolve.

Reply:
The only thing the loop does is to wait enough time to get the server in a defined (either right or wrong) state. So if there is an error different from ECONNREFUSED there is no need to stay waiting, there is no argue on that.

But I think there is no added value in reject the promise with an error either. This is because you will try to connect by yourself after the launch of the server, and you will be sucesfull or not, and get the right error in any case.

Anyway the promise could be rejected when there is an error detected, the change itself is trivial (although the specific tests are not that easy) but I cannot think of any use case in which this can be beneficial for the user of the module.